File: 백준/Bronze/2920. 음계/음계.py
# 1~8 한번씩 등장
# 이어지는 두 숫자는 증가하거나, 감소하거나
# 총 7번 증가 == 모두 증가 == 'ascending'
# 총 7번 감소 == 모두 감소 == 'descending'
# 그 외 == 'mixed'


keys = list(map(int, input().split()))
check = 0

for i in range(len(keys) - 1):
    if keys[i] < keys[i + 1]:
        check += 1
    else:
        check -= 1

if check == 7:
    print("ascending")
elif check == -7:
    print("descending")
else:
    print("mixed")

File: 백준/Bronze/2920. 음계/음계.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int check = 0;

        StringTokenizer st = new StringTokenizer(br.readLine());
        int first = Integer.parseInt(st.nextToken());
        for (int i = 1; i < 8; i++) {
            int second = Integer.parseInt(st.nextToken());
            if (second - first == 1 && check != -1) {
                check = 1;
            } else if (second - first == -1 && check != 1) {
                check = -1;
            } else {
                check = 0;
                break;
            }
            first = second;
        }

        switch (check) {
            case 1:
                System.out.println("ascending");
                break;
            case -1:
                System.out.println("descending");
                break;
            default:
                System.out.println("mixed");
                break;
        }

    }
}